patches/tModLoader/Terraria/ModLoader/Mod.Internals.cs

```diff
@@ -81,12 +81,40 @@ internal void PrepareAssets()
 			fileHandle = File?.Open();
 			RootContentSource = CreateDefaultContentSource();
 			Assets = new AssetRepository(Main.instance.Services.Get<AssetReaderCollection>(), new[] { RootContentSource }) {
-				AssetLoadFailHandler = Main.OnceFailedLoadingAnAsset
+				AssetLoadFailHandler = OnceFailedLoadingAnAsset
 			};
 		}
 
 		internal void TransferAllAssets() {
 			Assets.TransferAllAssets();
 		}
+
+		internal List<Exception> AssetExceptions = new List<Exception>();
+		internal void OnceFailedLoadingAnAsset(string assetPath, Exception e) {
+			if (!Main.gameMenu) {
```

Comment:
should just have a bool in the class, which is set in `TransferAllAssets` above like `initialTransferComplete = true` and we add to `AssetExceptions` until that bool is true.